lib/spack/spack/compilers/oneapi.py

```diff
@@ -138,6 +138,11 @@ def setup_custom_environment(self, pkg, env):
         if self.cxx:
             env.prepend_path("PATH", dirname(self.cxx))
 
+        # ensure that shared objects and libraries required by the compiler,
+        # e.g. libonnx, can be found succesfully
+        if self.cxx:
```

Comment:
```suggestion
        if self.cxx and pkg.spec.satisfies("%oneapi@:2024.1"):
```

This bug was fixed in 2024.2. Could you move the workaround from line 139 above here too? It is also fixed.

Reply:
Thank you @rscohn2, I have updated this as you requested.